homeassistant/components/sensor/tautulli.py

```diff
@@ -0,0 +1,133 @@
+"""
+A platform which allows you to get information from Tautulli.
+
+For more details about this component, please refer to the documentation at
+https://www.home-assistant.io/components/sensor.tautulli
+"""
+
+import logging
+import voluptuous as vol
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (CONF_API_KEY,
+                                 CONF_HOST,
+                                 CONF_MONITORED_VARIABLES,
+                                 CONF_PORT,
+                                 CONF_SSL,
+                                 STATE_UNAVAILABLE)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import (PLATFORM_SCHEMA)
+
+__version__ = '1.1.0'
+
+REQUIREMENTS = ['pytautulli==0.1.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MONITORED_USERS = 'monitored_users'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default='8181'): cv.string,
+    vol.Optional(CONF_SSL, default=False): cv.boolean,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=None):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_MONITORED_USERS, default=None):
+        vol.All(cv.ensure_list, [cv.string]),
+    })
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Create the sensor."""
+    api_key = config.get(CONF_API_KEY)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    monitored_variables = config.get(CONF_MONITORED_VARIABLES)
+    monitored_users = config.get(CONF_MONITORED_USERS)
+    ssl = config.get(CONF_SSL)
+    if ssl:
+        schema = 'https'
+    else:
+        schema = 'http'
+    add_devices([Tautulli(api_key, monitored_variables,
+                          host, port, monitored_users, schema)])
```

Comment:
Need to be renamed (-> `add_entities`).

We should check if the API key is valid before we setup the platform. Otherwise the users will end-up  with a non-functional platform in their setup.

To setup the platform when the host is down then `PlatformNotReady` can help.

Add `True` as second [parameter](https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#5-entity) to update the state on creation.